drivers/hmis/app/controllers/hmis/sessions_controller.rb

```diff
@@ -22,6 +22,16 @@ def new
 
   # POST /hmis/login
   def create
+    # If the account has been locked, show an appropriate message. Ideally this would happen after the password is
+    # validated however at that point there doesn't seem to be a way to differentiate failures for bad credentials vs
+    # those due to locked accounts without performing a second validation (which could trigger unintended lockouts)
```

Comment:
Just an FYI of something tangentially related: there is a bug with the devise-two-factor gem that causes the lockout count to be incremented twice from 1 attempt. as a result we always set the # failed attempts as double what it should be. 

https://github.com/greenriver/hmis-warehouse/blob/a99f2c8030b403ed3e2778a5f9918cb477b09762/drivers/hmis/spec/requests/hmis/sessions_controller_spec.rb#L55-L60